test/views/DeclarationSubmittedViewSpec.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views
+
+import play.twirl.api.HtmlFormat
+import views.html.DeclarationSubmittedView
+import views.behaviours.PanelViewBehaviours
+
+class DeclarationSubmittedViewSpec extends PanelViewBehaviours {
+
+  override val prefix: String = "declarationSubmitted"
+
+  override def view: HtmlFormat.Appendable =
+    injector.instanceOf[DeclarationSubmittedView].apply(mrn.toString, officeOfDestination)(fakeRequest, messages)
```

Comment:
Rather than using `officeOfDestination` here, could we use `Gen.oneOf(officeOfDestination, officeOfDestinationNoTel, officeOfDestinationNoNameTel, officeOfDestinationNoNameNoTel).sample.value`? Then we know the panel check, link check, h2 check etc. hold for each case